plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -299,20 +323,18 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
      * @param targetBlob new name of the blob in the same bucket
      */
     void moveBlob(String sourceBlobName, String targetBlobName) throws IOException {
-        final BlobId sourceBlobId = BlobId.of(bucket, sourceBlobName);
-        final BlobId targetBlobId = BlobId.of(bucket, targetBlobName);
+        final BlobId sourceBlobId = BlobId.of(bucketName, sourceBlobName);
+        final BlobId targetBlobId = BlobId.of(bucketName, targetBlobName);
         final CopyRequest request = CopyRequest.newBuilder()
                 .setSource(sourceBlobId)
                 .setTarget(targetBlobId)
                 .build();
-        SocketAccess.doPrivilegedVoidIOException(() -> {
-            // There's no atomic "move" in GCS so we need to copy and delete
-            storage.copy(request).getResult();
-            final boolean deleted = storage.delete(sourceBlobId);
-            if (deleted == false) {
-                throw new IOException("Failed to move source [" + sourceBlobName + "] to target [" + targetBlobName + "]");
-            }
-        });
+        // There's no atomic "move" in GCS so we need to copy and delete
+        storageAccessConsumer(storage -> storage.copy(request).getResult());
```

Comment:
This is where a safeClient() would be helpful, so that you have less chance that the underlying storage instance changed between the copy and delete calls

Reply:
Since google forces us to use two calls for the move op they should not be surprised if we do it over different clients...
Anyhow, using the present approach we could extend the lambda to:
```
storageAccess(storage -> {
  storage.copy(request).getResult();
  return storage.delete(sourceBlobId);
});
```
and keep it over a single client